src/battle_game/core/bullets/rocket.cpp

```diff
@@ -14,29 +16,88 @@ Rocket::Rocket(GameCore *core,
                glm::vec2 velocity)
     : Bullet(core, id, unit_id, player_id, position, rotation, damage_scale),
       velocity_(velocity) {
+  auto &units = game_core_->GetUnits();
+  auto player = game_core_->GetPlayer(player_id_);
+  auto &input_data = player->GetInputData();
+  auto WhereCursorIs = input_data.mouse_cursor_position;
+  float distance = 1e30;
+  for (auto &unit : units) {
+    if (unit.first == unit_id_) {
+      continue;
+    }
+    auto diff = WhereCursorIs - unit.second->GetPosition();
+    float length = sqrt(diff.x * diff.x + diff.y * diff.y);
+    if (length < distance) {
+      distance = length;
+      player_locked_ = unit.first;
+    }
+  }
 }
 
 void Rocket::Render() {
-  SetTransformation(position_, rotation_, glm::vec2{0.1f});
-  SetColor(game_core_->GetPlayerColor(player_id_));
-  SetTexture("../../textures/particle4.png");
+  SetTransformation(position_, rotation_, glm::vec2{1.0f});
+  SetTexture("../../textures/rocket.png");
   DrawModel(0);
 }
 
 void Rocket::Update() {
+  auto &units = game_core_->GetUnits();
+  if (units.find(player_locked_) == units.end()) {
+    game_core_->PushEventRemoveBullet(id_);
+    return;
+  }
+  if (harmful_ < 20)
+    harmful_ *= 1.02;
+  glm::vec2 diff;
+  for (auto &unit : units) {
+    if (unit.first == player_locked_) {
+      glm::vec2 nowpos = unit.second->GetPosition();
+      glm::vec2 whereIam = this->GetPosition();
+      diff = glm::vec2{nowpos.x - whereIam.x, nowpos.y - whereIam.y};
+      float length = sqrt(diff.x * diff.x + diff.y * diff.y);
+      diff = glm::vec2{0.25 * harmful_ * diff.x / length,
+                       0.25 * harmful_ * diff.y / length};
+    }
+  }
+  velocity_ = diff;
   position_ += velocity_ * kSecondPerTick;
+  float angle = 0;
+  if (diff.x < 0) {
+    if (diff.y < 0) {
+      angle = 3.141592653 / 2 + atan(abs(diff.y) / abs(diff.x));
```

Comment:
maybe you can use glm::pi<float>() instead, or you can add a macro somewhere public

Reply:
Thanks. Fixed